perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -849,27 +852,35 @@ public List<User> getUsersByAttributeValue(PerunSession sess, AttributeDefinitio
 		String value = "";
 		String operator = "=";
 		if (attributeDefinition.getType().equals(String.class.getName())) {
-			value = attributeValue.trim();
+			value = "'" + attributeValue.trim() + "'";
 			operator = "=";
 		} else if (attributeDefinition.getType().equals(Integer.class.getName())) {
-			value = attributeValue.trim();
+			value = "'" + attributeValue.trim() + "'";
 			operator = "=";
 		}  else if (attributeDefinition.getType().equals(Boolean.class.getName())) {
-			value = attributeValue.trim();
+			value = "'" + attributeValue.trim() + "'";
 			operator = "=";
 		} else if (attributeDefinition.getType().equals(ArrayList.class.getName())) {
-			value = "%" + attributeValue.trim() + "%";
+			// escape backslashes from delimiters escaping, use delimiters to match value
+			String escapedValue = AttributesManagerBlImpl.escapeListAttributeValue(attributeValue.trim()).replace("\\", "\\\\");
+			String firstElementMatch = escapedValue + LIST_DELIMITER + "%";
```

Comment:
We should always use parametrized queries (using JdbcTemplate's `named parameters` or `?`) instead of string concatenation to prevent SQL injections when passing input from users. 
I tested calling this method on my local instance with the parameters:
- attribute name `urn:perun:user:attribute-def:def:kerberosLogins` 
- attribute value `ok']);DROP TABLE user_attr_values CASCADE; commit;`

and the table was really dropped.